connector-common/src/main/java/com/redhat/cloud/notifications/connector/EngineToConnectorRouteBuilder.java

```diff
@@ -55,17 +58,26 @@ public void configure() throws Exception {
 
         onException(Throwable.class)
                 .onWhen(redeliveryPredicate::matches)
-                .handled(true)
+                .onWhen(continueOnErrorPredicate::matches)
+                .maximumRedeliveries(connectorConfig.getRedeliveryMaxAttempts())
+                .redeliveryDelay(connectorConfig.getRedeliveryDelay())
+                .retryAttemptedLogLevel(DEBUG)
+                .onRedelivery(redeliveryCounterProcessor)
+                .process(exceptionProcessor)
+                .continued(true);
+
+        onException(Throwable.class)
+                .onWhen(redeliveryPredicate::matches)
                 .maximumRedeliveries(connectorConfig.getRedeliveryMaxAttempts())
                 .redeliveryDelay(connectorConfig.getRedeliveryDelay())
                 .retryAttemptedLogLevel(DEBUG)
                 .onRedelivery(redeliveryCounterProcessor)
-                .process(exceptionProcessor);
+                .process(exceptionProcessor)
+                .handled(true);
 
         onException(Throwable.class)
-                .onWhen(redeliveryPredicate::doesNotMatch)
```

Comment:
According https://camel.apache.org/manual/exception-clause.html#_using_fine_grained_selection_using_onwhen_predicate we don't need the `doesNotMatch` predicate, because the `OnException` without `onWhen` condition will be considered only if other `OnException` with `onWhen` conditions failed